homeassistant/components/sia/__init__.py

```diff
@@ -0,0 +1,46 @@
+"""The sia integration."""
+import asyncio
+
+from homeassistant.components.alarm_control_panel import (
+    DOMAIN as ALARM_CONTROL_PANEL_DOMAIN,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+
+from .const import DOMAIN
+from .hub import SIAHub
+
+PLATFORMS = [ALARM_CONTROL_PANEL_DOMAIN]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up sia from a config entry."""
+    hub = SIAHub(hass, entry.data, entry.entry_id, entry.title)
+    await hub.async_setup_hub()
+
+    hass.data.setdefault(DOMAIN, {})
+    hass.data[DOMAIN][entry.entry_id] = hub
+
+    for component in PLATFORMS:
+        hass.async_create_task(
+            hass.config_entries.async_forward_entry_setup(entry, component)
+        )
+
+    await hub.sia_client.start(reuse_port=True)
```

Comment:
Start the client before forwarding the setup so we know if it fails.  Then raise `ConfigEntryNotReady` if you cannot connect or if auth is required `ConfigEntryAuthFailed`

Reply:
That's a good one, should I catch that on the config flow as well or does ha take care of that?